File: frontend/src/api/schedulingAssistant.ts
import { request } from './httpClient'

export interface InterpretedAppointment {
  patient_id: string | null
  attending_id: string | null
  room_id: string | null
  equipment_id: string | null
  start_time: string | null
  end_time: string | null
  notes: string | null
  warnings: string[]
}

export function interpretSchedulingRequest(token: string, text: string) {
  return request<InterpretedAppointment>('/scheduling/interpret', {
    method: 'POST',
    body: { text },
    token,
  })
}

File: backend/app/schemas/scheduling_assistant.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class InterpretRequestIn(BaseModel):
    text: str


class InterpretedAppointmentOut(BaseModel):
    patient_id: uuid.UUID | None = None
    attending_id: uuid.UUID | None = None
    room_id: uuid.UUID | None = None
    equipment_id: uuid.UUID | None = None
    start_time: datetime | None = None
    end_time: datetime | None = None
    notes: str | None = None
    warnings: list[str] = Field(default_factory=list)
